service/raft/raft_group0_client.cc

```diff
@@ -251,12 +251,22 @@ future<group0_guard> raft_group0_client::start_operation(seastar::abort_source*
     auto [upgrade_lock_holder, upgrade_state] = co_await get_group0_upgrade_state();
     switch (upgrade_state) {
         case group0_upgrade_state::use_post_raft_procedures: {
-            auto operation_holder = co_await get_units(_operation_mutex, 1);
+            auto operation_holder = co_await std::invoke([this, as] {
+                if (!as) {
```

Comment:
How much effort would it take to refactor `start_operation` to take `abort_source&` instead of `abort_source*`? (I wouldn't be surprised if all callsites already pass an `abort_source`)

Reply:
Yeah, will have a look maybe it is the case (so we could actually have just one call)

Reply:
See the comment - there is a place where we probably need to keep the pointer, but we can use work-around of creating a dummy abort source in that case ... and still looks worth the change as it simplifies the code significantly.